src/query/api/v1/handler/database/create.go

```diff
@@ -213,21 +213,25 @@ func (h *createHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// TODO(rartoul): Add test for NS exists.
-	_, nsExists := nsRegistry.Namespaces[namespaceRequest.Name]
-	if nsExists {
-		err := fmt.Errorf(
-			"unable to create namespace: %s because it already exists",
-			namespaceRequest.Name)
-		logger.Error("unable to create namespace", zap.Error(err))
-		xhttp.Error(w, err, http.StatusBadRequest)
-		return
+	for _, namespaceRequest := range namespaceRequests {
+		_, nsExists := nsRegistry.Namespaces[namespaceRequest.Name]
+		if nsExists {
```

Comment:
As an aside, do we still need this check? Seems like we don't really care if the NS already exists. Only issue would be if NS exists and we're trying to create it with different options, which we can check for.

Thoughts?

Reply:
would probably be less error prone to have a separate `update` api and to continue returning an error here.